### ISSUE TITLE
[DataGrid] Keyboard navigation doesn't work with pagination

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#1432]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

There are two broken behaviors:

1. The column header is only reachable with the navigation arrows when on the first page.
2. When the last recorded tabbable cell is not rendered on the current page, the grid isn't tabbable.

## Expected Behavior 🤔

1. The column header is reachable with the navigation arrows on any page.
2. There is always a cell that is tabbable.

## Steps to Reproduce 🕹

Steps:

1. Open https://material-ui.com/components/data-grid/#mit-version

## Context 🔦

This is a follow-up on #1289.

## Your Environment 🌎

v4.0.0-alpha.25
